File: lib/src/main/java/fr/o80/sample/lib/di/LibModule.java
package fr.o80.sample.lib.di;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import fr.o80.sample.lib.core.LibConfiguration;

/**
 * @author Olivier Perez
 */
@Module
public class LibModule {

    private final LibConfiguration configuration;

    public LibModule(LibConfiguration configuration) {
        this.configuration = configuration;
    }

    @Provides
    @Singleton
    public LibConfiguration provideLibConfiguration() {
        return configuration;
    }
}

File: lib/src/main/java/fr/o80/sample/lib/di/LibModule.kt
package fr.o80.sample.lib.di

import android.content.Context
import dagger.Module
import dagger.Provides
import fr.o80.sample.lib.core.LibConfiguration
import javax.inject.Singleton

/**
 * @author Olivier Perez
 */
@Module
class LibModule(private val configuration: LibConfiguration, private val context: Context) {

    @Provides
    @Singleton
    fun provideLibConfiguration(): LibConfiguration {
        return configuration
    }

    @Provides
    @Singleton
    fun provideContext(): Context {
        return context
    }

}
